src/main/java/io/zipcoder/Classroom.java

```diff
@@ -1,112 +1,99 @@
 package io.zipcoder;
 
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+
 public class Classroom {
     private Student[] students;
     // Helper fields
     private int studentsEnrolled;
     private int maxStudents;
 
-    /**
-     * Empty constructor.  You get to decide what max students should default to.
-     */
     public Classroom() {
-
+        this(30);
     }
 
-    /**
-     * Constructor where the caller defines wht maxStudents is.
-     * @param maxStudents
-     */
     public Classroom(int maxStudents) {
-
+        this.students = new Student[maxStudents];
     }
 
-    /**
-     * Constructor that builds a class from an existing Student array.
-     * @param students
-     */
     public Classroom(Student[] students) {
-
+        this.students = students;
     }
 
-    /**
-     * Adds a student to the class, and returns true.
-     * If you cannot add any more students, then return false and print an error statement.
-     * @param student
-     * @return
-     */
     public boolean addStudent(Student student) {
+        for (int i = 0; i < students.length; i++) {
+            if (students[i] == null) {
+                students[i] = student;
+                return true;
+            }
+        }
         return false;
     }
 
-    /**
-     * Returns the student with the firstName and lastName, and removes them from the array.
-     * If there is no student, return null.
-     * NOTE: You're going to want to move some students around in the array so that there are no empty spaces in the
-     * middle.
-     * @param firstName
-     * @param lastName
-     * @return
-     */
     public Student removeStudent(String firstName, String lastName) {
+        for (int i = 0; i < students.length; i++) {
+            if ((students[i].getFirstName() == firstName) && (students[i].getLastName() == lastName)) {
+                Student studentToBeRemoved = students[i];
+                students[i] = null;
+                return studentToBeRemoved;
+            }
+        }
         return null;
     }
 
-    /**
-     * Return the average score of all of the student's average scores.
-     */
     public double getClassAverage(){
-        return 100.0;
+        DecimalFormat decimalFormat = new DecimalFormat("###.#");
+        double sumOfAllAverages = 0;
+        int count = 0;
+        for (int i = 0; i < students.length; i++) {
+            if (students[i] != null) {
+                sumOfAllAverages += students[i].getAverage();
+                count++;
+            }
+        }
+        return Double.parseDouble(decimalFormat.format(sumOfAllAverages/count));
```

Comment:
i'm not liking the format to parse immediately, there are other, more optimal, ways to do this.

Reply:
Thanks.  Also, I realized that some of my tests were weird because I was pair coding with someone that wrote their methods to take 3 arguments when I wrote mine to take one object argument

Reply:
FYI: https://github.com/xt0fer/formatDopey